### PR TITLE
Docs and extra test for DH

### PR DESCRIPTION
**Pull request by hardbyte**
Suggested doc changes from https://github.com/pyca/cryptography/pull/2914

### CODE REVIEW

File: docs/hazmat/primitives/asymmetric/dh.rst [hunk 1]
```diff
@@ -6,92 +6,70 @@ Diffie-Hellman key exchange
...
-        The public value.
+    pn = dh.DHParameterNumbers(p, g)
+    parameters = pn.parameters(default_backend())
+    peer_public_numbers = dh.DHPublicNumbers(y, pn)
+    peer_public_key = peer_public_numbers.public_key(default_backend())
 
 
-.. class:: DHParameterNumbers(p, g)
+See also the :class:`~cryptography.hazmat.backends.interfaces.DHBackend`
+api for additional functionality.
```
Comment (palaviv) [inline docs/hazmat/primitives/asymmetric/dh.rst]: API

File: docs/hazmat/primitives/asymmetric/dh.rst [hunk 2]
```diff
@@ -6,92 +6,70 @@ Diffie-Hellman key exchange
...
+    :param generator: The :class:`int` to use as a generator. Often
+        2 or 5.
 
-    .. attribute:: g
+    :param key_size: The bit length of the prime modulus to generate.
 
-        :type: int
+    :param backend: A
+        :class:`~cryptography.hazmat.backends.interfaces.DHBackend`
+        provider.
```
Comment (palaviv) [inline docs/hazmat/primitives/asymmetric/dh.rst]: Don't use provider. See Issue #3073

File: src/cryptography/hazmat/backends/interfaces.py [hunk 3]
```diff
@@ -343,7 +343,8 @@ def load_dh_public_numbers(self, numbers):
     @abc.abstractmethod
     def load_dh_parameter_numbers(self, numbers):
         """
-        Returns a DHParameters provider.
+        Returns a DHParameters provider with parameters as a
```
Comment (palaviv) [inline src/cryptography/hazmat/backends/interfaces.py]: Let's remove the "provider".

**Comment by hardbyte** [comment]
Ok cheers, I didn't really like the work Provider - was just trying to stay consistent.

Hmm I hadn't read those [API Considerations](https://cryptography.io/en/latest/development/submitting-patches/#api-considerations) either, should the `DHBackend.dh_parameters_supported` return a bool or raise an exception?

More than happy to submit a pr to rewrite the `interfaces.py` docstrings in sphinx style if that is useful too.

**Comment by palaviv** [comment]
Thanks @hardbyte. Let's wait for someone from the cryptography project for CR before any more changes.

**Pull request merged**